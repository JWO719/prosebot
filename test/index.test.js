const { Application } = require('probot')
const appFn = require('..')

const payload = require('./fixtures/check_suite.requested.json')

describe('write-good-app', () => {
  let app, github, event

  beforeEach(() => {
    github = {
      pullRequests: {
        getFiles: jest.fn(() => Promise.resolve({ data: [
          { filename: 'added.md', status: 'added' },
          { filename: 'modified.md', status: 'modified' },
          { filename: 'deleted.md', status: 'deleted' }
        ] }))
      },
      repos: {
        getContent: jest.fn(() => Promise.resolve({ data: {
          content: Buffer.from('This here is some content!', 'utf8').toString('base64')
        }}))
      },
      checks: {
        create: jest.fn()
      }
    }

    app = new Application()
    app.load(appFn)
    app.auth = () => Promise.resolve(github)

    event = { name: 'check_suite', payload }
  })

  it('creates a check run', async () => {
    await app.receive(event)
    expect(github.checks.create).toHaveBeenCalled()
  })
})
