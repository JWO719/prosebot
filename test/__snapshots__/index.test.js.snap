// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`write-good-app creates a \`failing\` check run 1`] = `
Object {
  "conclusion": "failure",
  "head_branch": "pr",
  "head_sha": "296e2e88379d2ca5f211bac7bc25f78dc879236c",
  "name": "write-good-app",
  "output": Object {
    "annotations": Array [
      Object {
        "annotation_level": "warning",
        "end_line": 1,
        "message": "\\"So\\" adds no meaning",
        "path": "added.md",
        "start_line": 1,
      },
    ],
    "summary": "**1 suggestion** has been found in **1 file**.",
    "title": "See suggestions below.",
  },
  "owner": "JasonEtco",
  "repo": "tests",
}
`;

exports[`write-good-app creates a \`neutral\` check run if there are no files to check 1`] = `
Object {
  "conclusion": "neutral",
  "head_branch": "pr",
  "head_sha": "296e2e88379d2ca5f211bac7bc25f78dc879236c",
  "name": "write-good-app",
  "output": Object {
    "summary": "There were no \`.md\` or \`.txt\` files that needed checking.",
    "title": "No relevant files",
  },
  "owner": "JasonEtco",
  "repo": "tests",
}
`;

exports[`write-good-app creates a \`success\` check run 1`] = `
Object {
  "conclusion": "success",
  "head_branch": "pr",
  "head_sha": "296e2e88379d2ca5f211bac7bc25f78dc879236c",
  "name": "write-good-app",
  "output": Object {
    "annotations": Array [],
    "summary": "No issues have been found, great job!",
    "title": "No issues have been found!",
  },
  "owner": "JasonEtco",
  "repo": "tests",
}
`;
